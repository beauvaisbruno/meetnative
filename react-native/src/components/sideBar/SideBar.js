"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const native_base_1 = require("native-base");
const utils_1 = require("../../utils/utils");
const LogoutButton_1 = __importDefault(require("../profile/LogoutButton"));
const react_native_responsive_screen_1 = require("react-native-responsive-screen");
const react_native_auto_height_image_1 = __importDefault(require("react-native-auto-height-image"));
const custom_1 = __importDefault(require("../../config/native-base-theme/variables/custom"));
exports.renderRows = (data, navigation) => {
    return (React.createElement(native_base_1.ListItem, { onPress: () => {
            navigation.navigate(data.key);
            navigation.closeDrawer();
        }, button: true, testID: data.key },
        React.createElement(native_base_1.Text, { style: {
                color: navigation.state.routeName === data.key
                    ? custom_1.default.textGreyLight
                    : custom_1.default.textGrey
            } }, data.menuLabel)));
};
const SideBar = (props) => (React.createElement(native_base_1.Container, null,
    React.createElement(native_base_1.Content, null,
        React.createElement(react_native_auto_height_image_1.default, { source: require("../login/img/header.png"), width: react_native_responsive_screen_1.widthPercentageToDP("80%") }),
        React.createElement(native_base_1.List, { dataArray: utils_1.objsToArrayWithKey(props.routes), renderRow: data => exports.renderRows(data, props.navigation) }),
        React.createElement(LogoutButton_1.default, null))));
exports.default = SideBar;
