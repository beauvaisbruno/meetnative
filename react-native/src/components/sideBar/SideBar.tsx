import * as React from "react";
import { Container, Content, List, ListItem, Text } from "native-base";
import { objsToArrayWithKey } from "../../utils/utils";
import LogoutButton from "../profile/LogoutButton";
import { widthPercentageToDP as wp } from "react-native-responsive-screen";
import AutoHeightImage from "react-native-auto-height-image";
import { StyleProp, TextStyle } from "react-native";
import custom from "../../config/native-base-theme/variables/custom";
import {
  NavigationInjectedProps,
  NavigationScreenProp
} from "react-navigation";

export const renderRows = (
  data: { key: string; menuLabel: string },
  navigation: NavigationScreenProp<any, any>
) => {
  return (
    <ListItem
      onPress={() => {
        navigation.navigate(data.key);
        navigation.closeDrawer();
      }}
      button
      testID={data.key}
    >
      <Text
        style={{
          color:
            navigation.state.routeName === data.key
              ? custom.textGreyLight
              : custom.textGrey
        }}
      >
        {data.menuLabel}
      </Text>
    </ListItem>
  );
};

const SideBar = (
  props: { routes: { [x: string]: any } } & NavigationInjectedProps
) => (
  <Container>
    <Content>
      <AutoHeightImage
        source={require("../login/img/header.png")}
        width={wp("80%")}
      />
      <List
        dataArray={objsToArrayWithKey(props.routes)}
        renderRow={data => renderRows(data, props.navigation)}
      />
      <LogoutButton />
    </Content>
  </Container>
);
export default SideBar;
