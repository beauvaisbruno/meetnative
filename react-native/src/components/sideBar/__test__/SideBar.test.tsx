import * as React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import SideBar, { renderRows } from "../SideBar";

it("sidebar", () => {
  const navigate = jest.fn();

  const navigation = {
    closeDrawer: jest.fn(),
    navigate,
    state: { routeName: "routeName" }
  };

  const routes = {
    one: { menuLabel: "menuLabelOne" },
    two: { menuLabel: "menuLabelTwo" }
  };
  // @ts-ignore
  const tree = shallow(<SideBar routes={routes} navigation={navigation} />);
  expect(toJson(tree)).toMatchSnapshot();

  const dataArray: [{ key: string; menuLabel: string }] = tree
    .find("Styled(List)")
    .prop("dataArray");

  dataArray.forEach(data => {
    // @ts-ignore
    const listItems = shallow(renderRows(data, navigation));
    expect(toJson(listItems)).toMatchSnapshot();
    const submitBtn = listItems.findWhere(
      node => node.prop("testID") === data.key
    );
    submitBtn.props().onPress();
    expect(navigate).toBeCalledWith(data.key);
  });
});
