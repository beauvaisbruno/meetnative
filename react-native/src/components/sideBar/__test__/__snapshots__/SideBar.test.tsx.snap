// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sidebar 1`] = `
<Styled(Container)>
  <Styled(Content)>
    <ErrorableImage
      source={
        Object {
          "testUri": "../../../src/components/login/img/header.png",
        }
      }
      width={600}
    />
    <Styled(List)
      dataArray={
        Array [
          Object {
            "key": "one",
            "menuLabel": "menuLabelOne",
          },
          Object {
            "key": "two",
            "menuLabel": "menuLabelTwo",
          },
        ]
      }
      renderRow={[Function]}
    />
    <Connect(LogoutButton) />
  </Styled(Content)>
</Styled(Container)>
`;

exports[`sidebar 2`] = `
<ListItem
  button={true}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderBottomWidth": 0.5,
      "borderColor": "#c9c9c9",
      "flexDirection": "row",
      "marginLeft": 16,
      "paddingBottom": 13,
      "paddingRight": 16,
      "paddingTop": 13,
    }
  }
  testID="one"
>
  <Styled(Text)
    style={
      Object {
        "color": "#6d6d6d",
      }
    }
  >
    menuLabelOne
  </Styled(Text)>
</ListItem>
`;

exports[`sidebar 3`] = `
<ListItem
  button={true}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderBottomWidth": 0.5,
      "borderColor": "#c9c9c9",
      "flexDirection": "row",
      "marginLeft": 16,
      "paddingBottom": 13,
      "paddingRight": 16,
      "paddingTop": 13,
    }
  }
  testID="two"
>
  <Styled(Text)
    style={
      Object {
        "color": "#6d6d6d",
      }
    }
  >
    menuLabelTwo
  </Styled(Text)>
</ListItem>
`;
